core/unit_test/TestDefaultDeviceTypeInit.hpp

```diff
@@ -273,7 +281,7 @@ void check_correct_initialization(const Kokkos::InitArguments& argstruct) {
 
     int expected_device = argstruct.device_id;
     if (argstruct.device_id < 0) {
-      expected_device = 0;
+      expected_device = Kokkos::Impl::get_gpu(argstruct);
```

Comment:
Can't you just do
```suggestion
      expected_device = Kokkos::Cuda().cuda_device();
```
instead? I think we can ask the execution space for the device id it was initialized with.

Reply:
I will give that a try.

Reply:
Switching to:

```
expected_device = Kokkos::Cuda().cuda_device();
```

seems to work as well.  The updated commit is https://github.com/kokkos/kokkos/pull/3048/commits/0e51c8de02e5d9dd7bc87b99f2bb42ba7b7eae1d

Even though not needed, I would argue that you should keep the refactoring to pull out the `Kokkos::Impl::get_gpu()`.  That function could be unit tested in isolation much better and make this more solid and robust.

Reply:
This function would only be used in this context to initialize the device (id) properly and this id is available via `Kokkos::Cuda().cuda_device()`. Hence, I don't see a reason to test the former rather than the latter.
I don't feel strongly about pulling it out, though. This looks good enough for me.